STIPS/STIP-009.md

```diff
@@ -0,0 +1,743 @@
+)
+```
+
+TODO: more details here....
++ Set *setToken*, *factory* and *methodologist* public variables
++ Add allowed *_extensions* (these will be in set to *PENDING* state)
++ Add approved *_operators*
++ If *_allowedAssets* array is empty, set `anyAssetAllowed` to otherwise add allowedAssets to....
+
+> interactManager
+
+ONLY EXTENSION: Interact with a module registered on the SetToken
+
+```
+function interactManager(address _module, bytes calldata _data) external onlyExtension
+```
+
++ Require that *_module* is not the SetToken (to prevent operator from bypassing the extension interface)
++ Call `_module.functionCallWithValue(_data, 0)`
+
+> initializeExtension
+
+Initializes an added extension from PENDING to INITIALIZED state. An address can only
+enter a PENDING state if it is an enabled extension added by the manager. Only callable
+by the extension itself, hence msg.sender is the subject of update.
+
+```solidity
+function initializeExtension()
+```
+
++ Require that extension exists and it's state is *PENDING*
++ Set *extensionAllowlist[msg.sender]* to *INITIALIZED*
++ Add *msg.sender* to *extensions* array
+
+> addExtension
+
+ONLY OWNER: Add a new extension that the DelegatedManager can call
+
+```solidity
+function addExtensions(address[] memory _extensions)
+```
+
++ for each extension in _extensions
+  + require that extension state in *extensionAllowList* is *NONE* (has not already been added)
+  + set *extensionAllowlist[extension]* to PENDING
+  + emit *ExtensionAdded* event
+
+
+> removeExtensions
+
+ONLY OWNER: Remove an existing extension tracked by the DelegatedManager.
+
+```solidity
+function removeExtensions(address[] memory _extensions)
+```
+
++ for each extension in _extensions
+  + require that extension state in *extensionAllowList* is *INITIALZED*
+  + delete extension from *extensions* array
+  + set extension state to *NONE* in *extensionAllowList*
+  + call extensions's own *removeExtension* method for manager's setToken
+  + emit *ExtensionRemoved* event
+
+>
+
+### BaseGlobalExtension
+
+#### Modifiers
+
+> onlyAssetAllowList
+
+```solidity
+modifier onlyAssetAllowList(address memory _receiveAsset) {
+    require(manager.assetAllowList[_receiveAsset], "Must be allowed asset");
+    _;
+}
+```
+
+> onlyOwner
+
+```solidity
+modifier onlyOwner(ISetToken _setToken) {
+    require(msg.sender == _manager(_setToken).owner(), "Must be owner");
+    _;
+}
+```
+
+> onlyMethodologist
+
+```solidity
+modifier onlyMethodologist(ISetToken _setToken) {
+    require(msg.sender == _manager(_setToken).methodologist(), "Must be methodologist");
+    _;
+}
+```
+
+> onlyOperator
+
+```solidity
+modifier onlyOperator(ISetToken _setToken) {
+    require(_manager(_setToken).operatorAllowlist(msg.sender), "Must be approved operator");
+    _;
+}
+```
+
+> onlyManager
+
+```solidity
+modifier onlyManager(ISetToken _setToken) {
+    require(address(_manager(_setToken)) == msg.sender, "Manager must be sender");
+    _;
+}
+```
```

Comment:
Let's just note that we have `_manager` and `removeExtension` in here as unimplemented abstract functions